src/dynamicprompts/samplers/utils.py

```diff
@@ -50,3 +54,48 @@ def get_wildcard_not_found_fallback(
     res = SamplingResult(text=wrapped_wildcard)
     while True:
         yield res
+
+def replace_wildcard_variables(
+    command: WildcardCommand, 
+    *, 
+    context: SamplingContext,
+) -> WildcardCommand:    
+    prompt = pp.SkipTo(context.parser_config.variable_end)
+    variable_access = _configure_variable_access(
+        parser_config=context.parser_config,
+        prompt=prompt
+    )
+    variable_access.set_parse_action(_variable_replacement(context.variables))
+
+    wildcard_path_re = r"((?!" + re.escape(context.parser_config.wildcard_wrap) + r")[^(${}#])+"
+    wildcard_path = pp.Regex(wildcard_path_re).leave_whitespace()
+    wildcard = pp.Combine(pp.OneOrMore(variable_access | wildcard_path))("path")
+   
+    wildcard_result = wildcard.parse_string(command.wildcard)
+
+    return WildcardCommand(wildcard="".join(wildcard_result), sampling_method=command.sampling_method, variables=command.variables)
+    
+def _variable_replacement(variables: dict):
+    def var_replace(string, location, token):
```

Comment:
You could avoid the inner function with `functools.partial`:

```python
def _replace_variable(string, location, token, *, variables: dict): ...
```

...
```python
variable_access.set_parse_action(partial(_replace_variable, variables=context.variables))
```

Reply:
Updated.